README.md

```diff
@@ -37,5 +52,38 @@ export default class SomeComponent extends React.Component {
     }
   }
 }
-...
+
+class SomeComponent extend React.Component {
+  componentDidMount() {
+    this.cancelWhenUnmounted(
+      /* This will be automatically canceled if the observable does not
+       * onComplete or onError before the React component is unmounted.
+       */
+      fetchSomeData.subscribe( data => this.setstate({data}) )
+    );
+  }
+  render() {
+    return <CancelableComponent>
+    {(cancelWhenUnmounted, cancelAllSubscriptions) => {
+      // store the methods for cancelling, or they could be
+      // passed to a child component
+      this.cancelWhenUnmounted = cancelWhenUnmounted;
+      this.cancelAllSubscriptions = cancelAllSubscriptions;
+      return <div>My stuff</div>
+    }}
+    </CancelableComponent>
+  }
+}
+
+function YourModal({save, close}) {
+  return <div>
+    <button onClick={() => (save("data"), close())}>Save</button>
+    <button onClick={close}>cancel</button>
+  </div>
+}
+
+// This could easily be consumed by the Cancelable decorator or component
+cancelableModal(YourModal, propsToPass).subscribe(
```

Comment:
I think the documentation here could be improved to describe the properties of the observable that is being returned:
- The observable calls onCompleted when the modal is closed
- The observable calls onError whenever a rendering or other error occurs
- The observable may produce zero, one, or more values to onNext.
- The observable is a hot observable (I am 80% sure?)